src/System.Management.Automation/help/HelpProviderWithCache.cs

```diff
@@ -136,10 +135,10 @@ internal override IEnumerable<HelpInfo> SearchHelp(HelpRequest helpRequest, bool
                 foreach (string key in _helpCache.Keys)
                 {
                     if ((!searchOnlyContent && helpMatcher.IsMatch(key)) ||
-                        (searchOnlyContent && ((HelpInfo)_helpCache[key]).MatchPatternInContent(helpMatcher)))
+                        (searchOnlyContent && _helpCache[key].MatchPatternInContent(helpMatcher)))
                     {
                         countOfHelpInfoObjectsFound++;
-                        yield return (HelpInfo)_helpCache[key];
+                        yield return _helpCache[key];
```

Comment:
If we need the value why not enumerate by KeyValuePair?
```c#
foreach ((string key, HelpInfo value) in _helpCache)
```

Reply:
We do not need the value in all cases because of the condition above. I decided to avoid a wider refactoring.

Reply:
> KeyValuePair is a cheaper option than allocating class.

I don't see how this avoids any allocations. Additionally it would add perform a hash table lookup on every iteration, even in cases where it would not be required.

Reply:
> KeyValuePair is a cheaper option than allocating class.

I remember looking at this when I made the changes and decided it wasn't so straightforward.